Parsers/Find a book.js

```diff
@@ -0,0 +1,189 @@
+    output.success = false;
+    output.search = null;
+  } else if (invocation.indexOf('"') == -1) {
+    // Query invalid, let the user know
+    output.message = gs.getMessage('Book: An invalid query was provided. Please ensure query values are wrapped in double quotes for book lookup.');
+    output.success = false;
+    output.search = null;
+  } else {
+    // Prepare search output
+    output.search = {};
+    output.search.author = null;
+    output.search.title = null;
+    output.search.subject = null;
+    invocation = invocation.match(/(-\S)\s(".*?")/gmi);
+    for (var i = 0; i < invocation.length; i++) {
+      switch (invocation[i].charAt(1)) {
+        case 'a':
+          output.search.author = invocation[i].substring(invocation[i].indexOf('"'));
+          break;
+        case 't':
+          output.search.title = invocation[i].substring(invocation[i].indexOf('"'));
+          break;
+        case 's':
+          output.search.subject = invocation[i].substring(invocation[i].indexOf('"'));
+          break;
+        default:
+          break;
+      }
+    }
+    if (output.search.author === null && output.search.title === null && output.search.subject === null) {
+      output.message = gs.getMessage('Book: No supported flags were present in the command. Please call the command with `-help` for the supported flags and usage.');
+      output.success = false;
+      output.search = null;
+    } else {
+      output.message = '';
+      output.success = true;
+    }
+  }
+  return output;
+}
+
+function buildUrl(searchParams) {
+  var baseUrl = 'https://openlibrary.org/search.json?limit=1&q=';
+  var query = '';
+  if (searchParams.author !== null) {
+    query = query + ' author:' + searchParams.author;
+  }
+  if (searchParams.title !== null) {
+    query = query + ' title_suggest:' + searchParams.title;
+  }
+  if (searchParams.subject !== null) {
+    query = query + ' subject:' + searchParams.subject;
+  }
+  query = query.trim();
+
+  return baseUrl + encodeURIComponent(query);
+}
+
+function getBook(searchQuery) {
+  var output = {};
+  var searchMessage = new sn_ws.RESTMessageV2();
+  searchMessage.setEndpoint(searchQuery);
+  searchMessage.setHttpMethod('GET');
+  searchMessage.setRequestHeader('User-Agent', 'servicenow');
+  var searchResponse = searchMessage.execute();
+  if (searchResponse.getStatusCode() != 200) {
+    output.success = false;
+    output.message = gs.getMessage('Book: There was an issue querying OpenLibrary with your search. Please try again later, or reach out to @Astrid if this continues to occur.');
```

Comment:
if you change `@Astrid` to `<@astrid>` or `<@U013T46E4DA>` it'll be clickable to open a DM with you. up to you on whether you want them to put in the work themselves to get a hold of you 😂